src/app/ReadClient.cpp

```diff
@@ -77,95 +78,112 @@ void ReadClient::MoveToState(const ClientState aTargetState)
-
-        writer.Init(std::move(msgBuf));
-
-        err = request.Init(&writer);
-        SuccessOrExit(err);
-
-        if (aEventPathParamsListSize != 0 && apEventPathParamsList != nullptr)
+        EventPathList::Builder & eventPathListBuilder = request.CreateEventPathListBuilder();
+        EventPath::Builder eventPathBuilder           = eventPathListBuilder.CreateEventPathBuilder();
+        for (size_t eventIndex = 0; eventIndex < aEventPathParamsListSize; ++eventIndex)
         {
-            EventPathList::Builder & eventPathListBuilder = request.CreateEventPathListBuilder();
-            EventPath::Builder eventPathBuilder           = eventPathListBuilder.CreateEventPathBuilder();
-            for (size_t eventIndex = 0; eventIndex < aEventPathParamsListSize; ++eventIndex)
-            {
-                EventPathParams eventPath = apEventPathParamsList[eventIndex];
-                eventPathBuilder.NodeId(eventPath.mNodeId)
-                    .EventId(eventPath.mEventId)
-                    .EndpointId(eventPath.mEndpointId)
-                    .ClusterId(eventPath.mClusterId)
-                    .EndOfEventPath();
-                SuccessOrExit(err = eventPathBuilder.GetError());
-            }
+            EventPathParams eventPath = apEventPathParamsList[eventIndex];
+            eventPathBuilder.NodeId(eventPath.mNodeId)
+                .EventId(eventPath.mEventId)
+                .EndpointId(eventPath.mEndpointId)
+                .ClusterId(eventPath.mClusterId)
+                .EndOfEventPath();
+            SuccessOrExit(err = eventPathBuilder.GetError());
+        }
 
-            eventPathListBuilder.EndOfEventPathList();
-            SuccessOrExit(err = eventPathListBuilder.GetError());
+        eventPathListBuilder.EndOfEventPathList();
+        SuccessOrExit(err = eventPathListBuilder.GetError());
 
-            if (aEventNumber != 0)
-            {
-                // EventNumber is optional
-                request.EventNumber(aEventNumber);
-            }
+        if (aEventNumber != 0)
+        {
+            // EventNumber is optional
+            request.EventNumber(aEventNumber);
         }
+    }
 
-        if (aAttributePathParamsListSize != 0 && apAttributePathParamsList != nullptr)
+    if (aAttributePathParamsListSize != 0 && apAttributePathParamsList != nullptr)
+    {
+        AttributePathList::Builder attributePathListBuilder = request.CreateAttributePathListBuilder();
+        SuccessOrExit(err = attributePathListBuilder.GetError());
+        for (size_t index = 0; index < aAttributePathParamsListSize; index++)
         {
-            AttributePathList::Builder attributePathListBuilder = request.CreateAttributePathListBuilder();
-            SuccessOrExit(err = attributePathListBuilder.GetError());
-            for (size_t index = 0; index < aAttributePathParamsListSize; index++)
+            AttributePath::Builder attributePathBuilder = attributePathListBuilder.CreateAttributePathBuilder();
```

Comment:
`const &` could be used there as well to reference `apAttributePathParamsList[index]` since that seems to be used several times.